### DISCUSSION TITLE
Error in running SeisSol with FL=33

### DISCUSSION THREAD

**Discussion by jwjeremy** [SeisSol/SeisSol#681] [Q&A] [answered]
**Describe the bug**
I tried to run the seissol example [Northridge_FL33](https://github.com/SeisSol/Examples/tree/master/Northridge_FL33) in the supermuc and the seissol run got stuck in the following step `Info:  Initializing Fault, using a quadrature rule with  25  points.` : 

> Seissol output file: 
> ...
> Fri Sep 30 23:20:33, Info:  Initializing XDMF fault output.
> Fri Sep 30 23:20:33, Info:  Fault writer thread affinity: "-------789|0123456789|0123------|----------|----------|-----56789|0123456789|01--------|----------|------"
> Fri Sep 30 23:20:34, Info:  Initializing XDMF fault output. Done.
> Rank:        0 | Info    | Initializing element local matrices.
> Fri Sep 30 23:20:34, Info:  Initializing Fault, using a quadrature rule with  25  points.

In addition, I also tried to compute a similar `FrictionLaw= 33` simulation for the Tohoku earthquake and the simulation also stuck at the above step.

**Expected behavior**
I tried the same run within docker and the simulation is able to run. 

**To Reproduce**
_Steps to reproduce the behavior:_
- Follow the seissol example - [Northridge FL33](https://github.com/SeisSol/Examples/tree/master/Northridge_FL33) and update the parameter file with the latest SeisSol configuration

**Which version do you use?** 
- [master] Refactored DR Energy Output computations (#670)

**Which build settings do you use? Which compiler version do you use?**
- mpi compiler

**On which machine does your problem occur? If on a cluster: Which modules are loaded?**
- Supermuc
( - work well in seissol/training docker)

**Provide parameter/material files.**
- See [Northridge FL33](https://github.com/SeisSol/Examples/tree/master/Northridge_FL33)

**Screenshots/Console output**
- See the .out and .err output files. Many thanks~

**If you suspect a problem in the numerics/physics adds a screenshot of your output.**

**If you encounter any errors/warnings/... during execution please provide the console output.**

**Additional context
Add any other context about the problem here.**

Thanks in advance! :grinning:

[2367039.Northridge_FL33.err.txt](https://github.com/SeisSol/SeisSol/files/9687937/2367039.Northridge_FL33.err.txt)
[2367039.Northridge_FL33.out.txt](https://github.com/SeisSol/SeisSol/files/9687941/2367039.Northridge_FL33.out.txt)

**Comment by JeenaYun** [comment]
Hi Jeremy,

It seems like you have some problems with SeisSol in your SuperMUC.
I followed the example in my account and it works without being stuck.
You may try re-compiling SeisSol.

**Reply by jwjeremy** [reply]
Thanksss. My Supermuc seissol remains stuck at that stage. Hmmm

**Comment by Thomas-Ulrich** [comment, marked as answer]
Hi, 
I think there is probably an issue, and it is probable that the mesh of @JeenaYun and @jwjeremy are different (I already experienced gmsh giving slighly different mesh on different computer with the same geo file).
@jwjeremy could you then share your mesh file?
Also I note that you are using 16 nodes, but only 14 threads per nodes.
Please refer to https://seissol.readthedocs.io/en/latest/supermuc.html#running-seissol and do not change this part:
```
export OMP_NUM_THREADS=94
export OMP_PLACES="cores(47)"
```

**Reply by jwjeremy** [reply]
Hi Thomas, thank you so much! Please see the following link to the Mesh files. 
https://drive.google.com/drive/folders/1rg8ZzAnahj9cqBix5btP7RYNAeKeyUrb?usp=sharing

I have also tried the mesh by @JeenaYun however the problem persists.

**Reply by Thomas-Ulrich** [reply]
ok then maybe try her exe?

**Reply by jwjeremy** [reply]
I tried to run the Tohoku FL33 in her exe and the run is stuck on the  `Initializing Fault, using a quadrature rule with 25 points.` too.

I also attached the files and mesh used in the simulation. Please feel free to take a look and thanks so much!
https://drive.google.com/drive/folders/1Megx5GJ2u9ET1zMj-ohqqNbceBv6Jg5f?usp=sharing

**Reply by Thomas-Ulrich** [reply]
Hi, 
Tohoku_material.yaml is missing.

**Reply by Thomas-Ulrich** [reply]
But let's suppose I use this file:
https://github.com/SeisSol/Examples/blob/master/tpv29/tpv29_material.yaml
Then it is working for me:
```
Mon Oct 17 20:13:35, Info:  Initializing Fault, using a quadrature rule with  25  points.
Rank:        0 | Info    | DG initial condition projection...
Mon Oct 17 20:13:35, Info:  Using initial condition  "Zero" .
Rank:        0 | Info    | DG initial condition projection done.
Rank:        0 | Info    | <--------------------------------------------------------->
Rank:        0 | Info    | <     Start inioutput_SeisSol ...                         >
Rank:        0 | Info    | <--------------------------------------------------------->
Mon Oct 17 20:13:35, Info:  Initializing free surface integrator.
Mon Oct 17 20:13:35, Info:  Initializing free surface integrator. Done.
Mon Oct 17 20:13:36, Info:  Wave field writer thread affinity: "----------|----------|---3------|----------|----------|----------|----------|-1--------|----------|------"
Mon Oct 17 20:13:36, Info:  Initializing XDMF wave field output.
Mon Oct 17 20:13:36, Info:  Refinement Strategy is "Divide by 4"
Mon Oct 17 20:13:36, Info:  Refinement class initialized
Mon Oct 17 20:13:36, Info:  VariableSubsampler initialized
Mon Oct 17 20:13:37, Info:  High order output initialized
Mon Oct 17 20:13:37, Info:  Initializing XDMF wave field output. Done.
Mon Oct 17 20:13:37, Info:  Initializing free surface output.
Mon Oct 17 20:13:37, Info:  Free surface writer thread affinity: "----------|----------|---3------|----------|----------|----------|----------|-1--------|----------|------"
Mon Oct 17 20:13:37, Info:  Initializing free surface output. Done.
Mon Oct 17 20:13:37, Info:  Record points read from "offreceivers.dat"
Mon Oct 17 20:13:37, Info:  Number of record points = 6
Mon Oct 17 20:13:37, Info:  Finding meshIds for receivers...
Mon Oct 17 20:13:37, Info:  Cleaning possible double occurring receivers for MPI...
Mon Oct 17 20:13:37, Info:  Mapping receivers to LTS cells...
Mon Oct 17 20:13:37, Info:  Initializing energy output.
Rank:        0 | Info    | <--------------------------------------------------------->
Rank:        0 | Info    | <     Start calc_SeisSol ...                              >
Rank:        0 | Info    | <--------------------------------------------------------->
Mon Oct 17 20:13:37, Info:  Writing faultoutput at time 0.
Mon Oct 17 20:13:37, Info:  Writing faultoutput at time 0. Done.
Mon Oct 17 20:13:37, Info:  Waiting for last wave field.
Mon Oct 17 20:13:37, Info:  Writing wave field at time 0.
Mon Oct 17 20:13:37, Info:  Writing wave field at time 0. Done.
Mon Oct 17 20:13:37, Info:  Writing free surface at time 0.
Mon Oct 17 20:13:37, Info:  Writing free surface at time 0. Done.
Mon Oct 17 20:13:37, Info:  Writing energy output at time 0
Mon Oct 17 20:13:38, Info:  Writing energy output at time 0 Done.
Mon Oct 17 20:14:10, Info:  Writing energy output at time 0.5
Mon Oct 17 20:14:10, Info:  Elastic energy (total, % kinematic, % potential):  4.16106e+18  , 49.7801  , 50.2199
Mon Oct 17 20:14:10, Info:  Frictional work (total, % static, % radiated):  4.19804e+18  , 78.4355  , 21.5645
Mon Oct 17 20:14:10, Info:  Seismic moment (without plasticity): 3.99311e+21  Mw: 8.33087
```

**Reply by jwjeremy** [reply]
Oh, that's great news! I uploaded the material file used, which is a copy from the Northridge case. Thank you so much!

**Reply by Thomas-Ulrich** [reply]
Sorry, I m unable to reproduce this issue, even with the updated material file.

**Reply by jwjeremy** [reply]
I see, I am able to run seissol with the constant material and yet stuck on the 1D material file.

**Comment by Thomas-Ulrich** [comment]
You can try cloning a new version of seissol, compile and see if you still have the problem. Then if the problem is still there provide modules loaded, cmake compile command, hpc cluster used, launch script. 

Le 19 octobre 2022 19:21:01 GMT+02:00, Jeremy Wong ***@***.***> a écrit :

**Comment by JeenaYun** [comment]
If material file is the problem, then you may try changing 'depth' variable in the material file into 'zi'. Seems like the Northridge example is using 'zi'.

**Reply by jwjeremy** [reply]
@JeenaYun  @Thomas-Ulrich  It works now! Thank you so much! I didn't notice there is an update in the examples xd.

**Reply by AliceGabriel** [reply]
Tagging @Yonatan298 to point to this discussion.
